src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -0,0 +1,202 @@
+    // "$apiUrl/comic/update/list/1/$page" is same content
+    override fun latestUpdatesRequest(page: Int): Request =
+        GET("$apiUrl/comic/update/list/0/$page")
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val mangas = response.parseAs<ResponseDto<List<PageItemDto>>>().data!!
+        return MangasPage(mangas.map { it.toSManga() }, true)
+    }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            EditTextPreference(screen.context).apply {
+                key = "USERNAME"
+                title = "用户名"
+                setOnPreferenceChangeListener { _, newValue ->
+                    preferences.edit().putString("USERNAME", newValue as String).commit()
+                    true
+                }
+            }.let(screen::addPreference)
+
+            EditTextPreference(screen.context).apply {
+                key = "PASSWORD"
+                title = "密码"
+                setOnPreferenceChangeListener { _, newValue ->
+                    preferences.edit().putString("PASSWORD", newValue as String).commit()
+                    true
+                }
+            }.let(screen::addPreference)
+        }
+
+        SwitchPreferenceCompat(screen.context).apply {
+            title = "更新Token"
+            summary = "填写用户名及密码设置后，必须手动点击此选项来登录以更新Token"
+            setOnPreferenceChangeListener { _, _ ->
+                thread {
+                    try {
+                        // TODO: showToast not work on mihon but work on yokai/tachiyomiJ2K
+                        var t: String = preferences.getString("TOKEN", "")!!
+                        if (t.isNullOrBlank() || isExpired(t)) {
+                            val username = preferences.getString("USERNAME", "")!!
+                            val password = preferences.getString("PASSWORD", "")!!
+                            if (username.isEmpty() || password.isEmpty()) {
+                                showToast(screen.context, "用户名或者密码未输入，请填写完整后再登录")
+                            } else {
+                                t = getToken(username, password)
+                                if (t.isNullOrBlank()) {
+                                    showToast(screen.context, "登录失败，请检查用户名/密码是否正确")
+                                } else {
+                                    preferences.edit().putString("TOKEN", t).apply()
+                                    apiHeaders.newBuilder().setToken(t).build()
+                                    showToast(screen.context, "登录成功，Token已经成功更新")
+                                }
+                            }
+                        } else {
+                            showToast(screen.context, "Token有效，无需更新")
+                        }
+                    } catch (e: Throwable) {
+                        Log.e("Zaimanhua", "failed to update token", e)
+                    }
+                }
+                false
```

Comment:
consider setting the token in an interceptor

on an api request, if username and password exist, get token and save it in pref and continue to use it for subsequent requests